libbpf-tools/oomkill.bpf.c

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2022 Tencent
+#include <vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+
+#include "oomkill.h"
+
+struct {
+	__uint(type, BPF_MAP_TYPE_PERF_EVENT_ARRAY);
+	__uint(key_size, sizeof(u32));
+	__uint(value_size, sizeof(u32));
+} events SEC(".maps");
+
+SEC("kprobe/oom_kill_process")
+int BPF_KPROBE(oom_kill_process, struct oom_control *oc, const char *message) {
+	unsigned long totalpages;
+	struct task_struct *p = NULL;
+	p = BPF_CORE_READ(oc, chosen);
+	struct data_t data = {};
+	u32 pid = bpf_get_current_pid_tgid() >> 32;
+	data.fpid = pid;
+	data.tpid = BPF_CORE_READ(p, pid);
+	data.pages = BPF_CORE_READ(oc, totalpages);
+	bpf_get_current_comm(&data.fcomm, sizeof(data.fcomm));
+	bpf_probe_read_kernel(&data.tcomm, sizeof(data.tcomm), BPF_CORE_READ(p, comm));
+	bpf_perf_event_output(ctx, &events, BPF_F_CURRENT_CPU, &data, sizeof(data));
+	return 0;
+}
```

Comment:
* Declare variables first and leave an empty line below.
* `totalpages` is not used, please remove it.
* No need to zero-initialized `p` and `data`.